ocf_datapipes/select/select_time_slice_nwp.py

```diff
@@ -78,23 +88,59 @@ def __iter__(self) -> Union[xr.DataArray, xr.Dataset]:
                 t0_available = t0
 
             # Forecasts made up to and including t0
-            xr_available = xr_data.sel(init_time_utc=slice(None, t0_available))
+            available_init_times = xr_data.init_time_utc.sel(
+                init_time_utc=slice(None, t0_available)
+            )
 
-            init_times = xr_available.sel(
+            # Find the most recent available init times for all target times
+            selected_init_times = available_init_times.sel(
                 init_time_utc=target_times,
                 method="ffill",  # forward fill from init times to target times
-            ).init_time_utc.values
+            ).values
 
-            steps = target_times - init_times
+            # Find the required steps for all target times
+            steps = target_times - selected_init_times
 
             # We want one timestep for each target_time_hourly (obviously!) If we simply do
             # nwp.sel(init_time=init_times, step=steps) then we'll get the *product* of
             # init_times and steps, which is not what # we want! Instead, we use xarray's
             # vectorized-indexing mode by using a DataArray indexer.  See the last example here:
             # https://docs.xarray.dev/en/latest/user-guide/indexing.html#more-advanced-indexing
             coords = {"target_time_utc": target_times}
-            init_time_indexer = xr.DataArray(init_times, coords=coords)
+            init_time_indexer = xr.DataArray(selected_init_times, coords=coords)
             step_indexer = xr.DataArray(steps, coords=coords)
-            xr_sel = xr_available.sel(step=step_indexer, init_time_utc=init_time_indexer)
+
+            # Slice out the data which does not need to be diffed
+            xr_non_accum = xr_data.sel({self.channel_dim_name: non_accum_channels})
+            xr_non_accum = xr_non_accum.sel(step=step_indexer, init_time_utc=init_time_indexer)
+
+            if len(accum_channels) == 0:
+                xr_sel = xr_non_accum
+
+            else:
+                # First minimise the size of the dataset we are diffing
```

Comment:
worth putting some sort of logging step here?

Reply:
This would cause a message for every sample which would clutter the logs. But I'll add something in before the loop